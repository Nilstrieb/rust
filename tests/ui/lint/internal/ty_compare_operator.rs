// compile-flags: -Z unstable-options

#![feature(rustc_attrs)]
#![forbid(rustc::ty_compare_operator)]

mod ty1 {
    #[derive(PartialEq, PartialOrd)]
    #[rustc_diagnostic_item = "Ty"]
    pub struct Ty;
}

mod ty2 {
    #[derive(PartialEq, PartialOrd)]
    pub struct Ty;
}

fn main() {
    let _ = ty1::Ty == ty1::Ty;
    //~^ ERROR using the a comparison operator on `Ty`
    let _ = ty1::Ty != ty1::Ty;
    //~^ ERROR using the a comparison operator on `Ty`
    let _ = ty1::Ty < ty1::Ty;
    //~^ ERROR using the a comparison operator on `Ty`
    let _ = ty1::Ty <= ty1::Ty;
    //~^ ERROR using the a comparison operator on `Ty`
    let _ = ty1::Ty > ty1::Ty;
    //~^ ERROR using the a comparison operator on `Ty`
    let _ = ty1::Ty >= ty1::Ty;
    //~^ ERROR using the a comparison operator on `Ty`

    let _ = ty2::Ty == ty2::Ty;
    let _ = ty2::Ty != ty2::Ty;
    let _ = ty2::Ty < ty2::Ty;
    let _ = ty2::Ty <= ty2::Ty;
    let _ = ty2::Ty > ty2::Ty;
    let _ = ty2::Ty >= ty2::Ty;
}
