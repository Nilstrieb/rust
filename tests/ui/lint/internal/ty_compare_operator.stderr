error: using the a comparison operator on `Ty`
  --> $DIR/ty_compare_operator.rs:18:21
   |
LL |     let _ = ty1::Ty == ty1::Ty;
   |                     ^^
   |
   = note: this does probably not what you want as it does not handle inference variables and more
   = note: it's also not recommended to use it for diagnostics
   = help: for more information, see https://rustc-dev-guide.rust-lang.org/ty.html#comparing-types
note: the lint level is defined here
  --> $DIR/ty_compare_operator.rs:4:11
   |
LL | #![forbid(rustc::ty_compare_operator)]
   |           ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: using the a comparison operator on `Ty`
  --> $DIR/ty_compare_operator.rs:20:21
   |
LL |     let _ = ty1::Ty != ty1::Ty;
   |                     ^^
   |
   = note: this does probably not what you want as it does not handle inference variables and more
   = note: it's also not recommended to use it for diagnostics
   = help: for more information, see https://rustc-dev-guide.rust-lang.org/ty.html#comparing-types

error: using the a comparison operator on `Ty`
  --> $DIR/ty_compare_operator.rs:22:21
   |
LL |     let _ = ty1::Ty < ty1::Ty;
   |                     ^
   |
   = note: this does probably not what you want as it does not handle inference variables and more
   = note: it's also not recommended to use it for diagnostics
   = help: for more information, see https://rustc-dev-guide.rust-lang.org/ty.html#comparing-types

error: using the a comparison operator on `Ty`
  --> $DIR/ty_compare_operator.rs:24:21
   |
LL |     let _ = ty1::Ty <= ty1::Ty;
   |                     ^^
   |
   = note: this does probably not what you want as it does not handle inference variables and more
   = note: it's also not recommended to use it for diagnostics
   = help: for more information, see https://rustc-dev-guide.rust-lang.org/ty.html#comparing-types

error: using the a comparison operator on `Ty`
  --> $DIR/ty_compare_operator.rs:26:21
   |
LL |     let _ = ty1::Ty > ty1::Ty;
   |                     ^
   |
   = note: this does probably not what you want as it does not handle inference variables and more
   = note: it's also not recommended to use it for diagnostics
   = help: for more information, see https://rustc-dev-guide.rust-lang.org/ty.html#comparing-types

error: using the a comparison operator on `Ty`
  --> $DIR/ty_compare_operator.rs:28:21
   |
LL |     let _ = ty1::Ty >= ty1::Ty;
   |                     ^^
   |
   = note: this does probably not what you want as it does not handle inference variables and more
   = note: it's also not recommended to use it for diagnostics
   = help: for more information, see https://rustc-dev-guide.rust-lang.org/ty.html#comparing-types

error: aborting due to 6 previous errors

